File: snap_crescent_mobile-flutter/lib/models/location.dart
import 'package:snap_crescent/models/base_model.dart';

class Location extends BaseUiBean {
  double? longitude;
  double? latitude;
  String? country;
  String? state;
  String? city;
  String? town;
  String? postcode;

  Location(
      {
      bean,
      this.longitude,
      this.latitude,
      this.country,
      this.state,
      this.city,
      this.town})
      : super(
            id: bean.id,
            version: bean.version,
            creationDatetime: bean.creationDatetime,
            lastModifiedDatetime: bean.lastModifiedDatetime,
            active: bean.active);

  factory Location.fromJson(Map<String, dynamic> json) {
    return Location(
        bean: BaseUiBean.fromJson(json),
        longitude: json['longitude'],
        latitude: json['latitude'],
        country: json['country'],
        state: json['state'],
        city: json['city'],
        town: json['town']);
  }
}

File: snapcrescent/src/main/java/com/snapcrescent/location/Location.java
package com.snapcrescent.location;

import jakarta.persistence.Entity;

import com.snapcrescent.common.BaseEntity;

import lombok.Data;
import lombok.EqualsAndHashCode;

@Entity
@Data
@EqualsAndHashCode(callSuper = false)
public class Location extends BaseEntity {

	private static final long serialVersionUID = -5153885822493234594L;

	private Double longitude;
	private Double latitude;
	private String country;
	private String state;
	private String city;
	private String town;
	private String postcode;

}
